libraries/botbuilder-dialogs-adaptive/src/input/confirmInput.ts

```diff
@@ -218,4 +216,26 @@ export class ConfirmInput extends InputDialog implements ConfirmInputConfigurati
 
         return culture;
     }
+
+    private async getConfirmChoices(dc: DialogContext, defaults: (string | Choice)[]): Promise<ChoiceSet> {
+        let confirmChoices: ChoiceSet;
+        if (this.confirmChoices != null) {
+            if (
+                this.confirmChoices.expressionText != null &&
+                this.confirmChoices.expressionText.trimLeft().startsWith('${')
+            ) {
+                // use TemplateInterface to bind (aka LG)
+                confirmChoices = await new ChoiceSet(this.confirmChoices.expressionText).bind(dc, dc.state);
```

Comment:
AFAIK constructors can't be async, so you shouldn't need `await` here.

Reply:
the `await` is wating for the "bind" function. Anyway, it is better to separate them to avoid ambiguity

Reply:
The `await` is to wait for the 'bind' function, not for the constructor